railgun/hack/generators/manifests/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// -----------------------------------------------------------------------------
+// Vars
+// -----------------------------------------------------------------------------
+
+var (
+	// directory that will be used for walking the filesystem to find CRDs.
+	directory string
+
+	// patchfile is the patch that will be used for each CRD file
+	patchfile = "config/crd/patches/upgrade_compat.yaml"
```

Comment:
Is this patch or upgrade ? the function in this PR more like crd/upgrade

Reply:
this is a patch which enables CRD upgrade compatibility. "patch" in this case is a specific type of YAML file which this file conforms to.

Reply:
Generically, software patch means operation fix, update, improve. Aren't you  doing both version compatible.

Reply:
I'm sorry I may need further help to understand the problem, could you please add a suggestion of the name you would prefer?

Reply:
upgrade
migrate

Reply:
I'm sorry, could you give a very prescriptive suggestion, such as suggesting the full path to the file as a github suggestion? I'm afraid it's still not clear to me _where_ to change this language.

Reply:
Q: "could you please add a suggestion of the name you"
A: upgrade, migrate

Reply:
Drive-by comment to add some more context and a specific proposed action stemming from it:

@ccfishk, "patch" here refers to the act of "patching" a YAML file with a "YAML patch". See [jsonpatch](http://jsonpatch.com/) as an example use of the term "patch" for overwriting a part of a JSON object.

This PR implements a program that takes the v1 CRDs generated by kubebuilder and **patches** them with a change that makes them suitable as an upgrade/migration from v1beta1 CRDs.

In other words, kubebuilder generates v1 CRDs that don't apply cleanly over v1beta1 CRDs.

The v1 CRDs patched with this YAML patch, however, can be applied over v1beta1 CRDs.

Does that make sense now? In other words, I think that the word "patch" here is appropriate and we should keep it as-is. Because "patching" does not refer to "patched software" but to "a YAML file with a patch applied to it".